### ISSUE TITLE
PoET: skip validating PoET proof if it already exists

### ISSUE THREAD

**Issue by countvonzero** [spacemeshos/go-spacemesh#3153]
## Description

we validate every syntactically correct PoET proof regardless whether it's already present locally.

from Noam:
```
Validating a PoET proof is definitely more expensive than calculating a Sha256 sum of the proof.
It involves calculating many smaller Sha256 sums (I think the lower bound for the number of
hashed bytes is proof_size*log(number_of_poet_iterations)), as well hashes for the leaves. 
So the number of hashed bytes for validation is a multiple of the proof, likely 10x or more.
```

## Actual Behavior

validate regardless

## Expected Behavior

only validate when needed

## Additional Resources 

i made an attempt with https://github.com/spacemeshos/go-spacemesh/pull/3143 and turned out that apptest failed because rejecting duplicate msg is considered as a p2p publish failure.

**Comment by dshulyak** [comment]
> i made an attempt with https://github.com/spacemeshos/go-spacemesh/pull/3143 and turned out that apptest failed because rejecting duplicate msg is considered as a p2p publish failure.

@countvonzero do you recall what was the actual problem? there is a similar pattern in other parts (for sure in atxs), if object exists it is being ignored in p2p gossip, and handled as a no-op for sync

**Comment by countvonzero** [comment]
a few things: 
- this line https://github.com/spacemeshos/go-spacemesh/blob/9202e3e32945e769df72c74bfb9253dd7cb11a22/activation/poetlistener.go#L42

made one think it is ignored in p2p gossip, but it's not because the error wrapper is not properly wrapped here
https://github.com/spacemeshos/go-spacemesh/blob/9202e3e32945e769df72c74bfb9253dd7cb11a22/activation/poetdb.go#L92
(%v vs %w). so duplicate poet proof always got re-broadcasted in p2p.

- in apptest https://github.com/spacemeshos/go-spacemesh/blob/9202e3e32945e769df72c74bfb9253dd7cb11a22/cmd/node/app_test.go#L155
we fail fast when poet failed to publish a message. and when i added the exist check, `ValidationIgnore` is considered an error. so the test fails.

- in slack discussion https://spacemesh.slack.com/archives/C89QJJY3G/p1646020377694479 it was also mentioned the primary key currently used for poet proof isn't completely correct. it misses the round id. from Noam's message, it seems we should not use Ref() as the primary key and should create a separate ID that hashes that entire message.

**Comment by dshulyak** [comment]
weird, github stopped closing issues

**Comment by noamnelke** [comment]
> weird, github stopped closing issues

The reason it wasn't closed is that this issue was mentioned in #3191 (which was closed, but not merged) but it was never mentioned in #3143 which actually fixes this issue (and WAS merged).

As an aside, in our workflow it's bors that closes referenced issues when it merges PRs.
